app/models/oauth_token.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class OauthToken < ApplicationRecord
+  OAUTH_APPLICATION_NAME = 'ic-rails-huey-toby'
+  OAUTH_APPLICATION_REDIRECT_URI = 'urn:ietf:wg:oauth:2.0:oob'
+
+  class << self
+    # https://github.com/doorkeeper-gem/doorkeeper/issues/846#issuecomment-230297646
+    def generate_access_token(user)
+      Doorkeeper::AccessToken.create(
+        application_id: generate_application.id,
+        resource_owner_id: user.id,
+        refresh_token: Doorkeeper::OAuth::Helpers::UniqueToken.generate,
+        expires_in: Doorkeeper.configuration.access_token_expires_in.to_i,
+        scopes: Doorkeeper.configuration.default_scopes
+      )
+    end
+
+    private
+
+    def generate_application
+      Doorkeeper::Application.find_or_create_by(name: OAUTH_APPLICATION_NAME,
```

Comment:
The OAuth application should be generated via the seed file when setting up the application. It should rely on having a user trying to authenticate for the first time 🙈 Also. ensuring that the OAuth application exists upon deployment is needed. So `find_or_create_by` is not the right call here. Also, the method name does not describe what it does.

Reply:
@olivierobert Thanks for your comment. I added the default oauth application to the seed file, besides that, the new migration file was also added to make sure a new record will be added to oauth_application table on deployment. So in the `generate_access_token` method, the first record of `Doorkeeper::Application` will be always used for generating access token. I am missing anything? 😄

Reply:
All good now 👍